src/Components/Components/test/Auth/AuthorizeRouteViewTest.cs

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components
+{
+    public class AuthorizeRouteViewTest
```

Comment:
The tests in this file are very much on the "integration test" end of the scale, and were quite hard to write. When you have a component that renders more components, it's pretty arbitrary how far you go with testing the overall result.

I wasn't happy with an uncompromisingly unit-testy approach, because simply asserting that we render a specific other component with some params is just a restatement of the implementation and doesn't prove anything about whether it produces the right number of render batches, etc. So I've tried to strike a balance where these tests assert about the most relevant net rendering effects only.

When we get to designing a proper unit testing system for components, these would be great case studies. The developer writing tests *might* want to make assertions about the number of batches and the combination of descendant components and their params, *or* they might only want to assert about the resulting document shape that you get when combining all the rendering effects.